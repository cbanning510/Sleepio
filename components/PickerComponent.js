import React from 'react';
import {StyleSheet} from 'react-native';

import {Picker} from '@react-native-picker/picker';

const PickerComponent = ({callback, items, value, setValue}) => {
  return (
    <Picker
      style={styles.container}
      selectedValue={value}
      onValueChange={(itemValue, itemIndex) => {
        setValue(itemValue);
        callback();
      }}>
      {items.map((d, i) => (
        <Picker.Item
          color="white"
          key={d + i}
          label={d.toString()}
          value={d.toString()}
        />
      ))}
    </Picker>
  );
};

const styles = StyleSheet.create({
  container: {width: '100%', textAlign: 'left', backgroundColor: 'dodgerblue'},
});

export default PickerComponent;
