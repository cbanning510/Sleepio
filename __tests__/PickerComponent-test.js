import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PickerComponent from '../components/PickerComponent';

const callback = () => console.log('hello Picker');
const items = [0, 0.5, 1, 1.5, 2];
const value = 1;
const setValue = () => {};

it('renders correctly', () => {
  const tree = renderer.create(
    <PickerComponent
      items={items}
      value={value}
      callback={callback}
      setValue={setValue}
    />,
  );
  expect(tree).toMatchSnapshot();
});
