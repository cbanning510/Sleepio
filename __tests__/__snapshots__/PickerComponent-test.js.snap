// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "dodgerblue",
      "textAlign": "left",
      "width": "100%",
    }
  }
>
  <RNCPicker
    items={
      Array [
        Object {
          "label": "0",
          "testID": undefined,
          "textColor": 4294967295,
          "value": "0",
        },
        Object {
          "label": "0.5",
          "testID": undefined,
          "textColor": 4294967295,
          "value": "0.5",
        },
        Object {
          "label": "1",
          "testID": undefined,
          "textColor": 4294967295,
          "value": "1",
        },
        Object {
          "label": "1.5",
          "testID": undefined,
          "textColor": 4294967295,
          "value": "1.5",
        },
        Object {
          "label": "2",
          "testID": undefined,
          "textColor": 4294967295,
          "value": "2",
        },
      ]
    }
    numberOfLines={1}
    onChange={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
